### DISCUSSION TITLE
Property misspelled or does not exist, not sure what am I missing

### DISCUSSION THREAD

**Discussion by pr4bh4sh** [detekt/detekt#5304] [Q&A] [answered]
When I run `./gradlew detekt` I get below error 

```bash
❯ ./gradlew clean

BUILD SUCCESSFUL in 1s
2 actionable tasks: 2 executed
❯ ./gradlew detekt

> Task :detekt-custom-rules:detekt FAILED
Property 'formatting' is misspelled or does not exist.
Property 'detekt-custom-rules' is misspelled or does not exist.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':detekt-custom-rules:detekt'.
> Run failed with 2 invalid config properties.
        - Property 'formatting' is misspelled or does not exist.
        - Property 'detekt-custom-rules' is misspelled or does not exist.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
3 actionable tasks: 3 executed
❯ ./gradlew detekt

> Task :detekt FAILED
Property 'detekt-custom-rules' is misspelled or does not exist.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':detekt'.
> Run failed with 1 invalid config property.
        - Property 'detekt-custom-rules' is misspelled or does not exist.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
5 actionable tasks: 3 executed, 2 up-to-date
 ```

I'm not sure what am I doing wrong as I get a different error on the same command.

Could someone please have a look and help me?
https://github.com/pr4bh4sh/detekt-demo

**Comment by cortinico** [comment]
> Execution failed for task ':detekt-custom-rules:detekt'.
> Run failed with 2 invalid config properties.
>        - Property 'formatting' is misspelled or does not exist.
>        - Property 'detekt-custom-rules' is misspelled or does not exist.

The error message says that you have two keys in your config which are not valid: `formatting` and `detekt-custom-rules`. The `formatting` one is most likely related to not importing the detekt formatting plugin correctly.

The `'detekt-custom-rules'` similarly. Are you trying to develop a custom rule by any chance?

**Reply by adavis** [reply]
> The formatting one is most likely related to not importing the detekt formatting plugin correctly.

How can we properly import this plugin? Thanks!

**Reply by BraisGabin** [reply]
https://detekt.dev//docs/intro#adding-more-rule-sets here it is described. If it's not clear, please open an issue so we can improve our documentation :)

**Reply by houshuo521** [reply]
why config.yml need depenceny this formating plugin? This is not convenient

**Reply by BraisGabin** [reply]
It doesn't need it. If you don't use that plugin you should not add it to your config and that's it.

**Reply by houshuo521** [reply]
i want to exec “java -jar” by CI，but dont change build.gradle,what should i to do ?
Now i use config.yml by order “ java -jar detekt-cli-1.23.6-all.jar --config config_detekt_config.yml --input app/src/main/java/ --report html:build/reports/detekt/detekt.html ”，then error message indicate “Run failed with 1 invalid config property.
- Property 'formatting' is misspelled or does not exist.

**Reply by houshuo521** [reply]
i fix this issue by order“  java -classpath detekt-cli-1.23.6-all.jar:detekt-formatting-1.23.6.jar io.gitlab.arturbosch.detekt.cli.Main --config config_detekt_config.yml --input app/src/main/java/com/MainActivity.kt    --report html:build/reports/detekt/detekt1.html
”.   thanks !

**Comment by pr4bh4sh** [comment]
Thanks for taking a look @cortinico. Really appreciate the work you have done on detekt and providing support here for noobs like me :) . I'm coming from https://github.com/sds/overcommit and https://github.com/rubocop/rubocop, migrating to kotlin and detekt for good.

I have imported the detekt formatting plugin [here](https://github.com/pr4bh4sh/detekt-demo/blob/master/build.gradle.kts#L68). Not sure what could be wrong with it. 

As mentioned in the previous comment when I ran the `./gradlew detekt` a second time the error about the 'formatting' misspelling went away. There was no change in the code/configuration between running the command. 

Yes, I plan to develop some custom rules in future.
I have added the detektProject dependency for custom rules [here](https://github.com/pr4bh4sh/detekt-demo/blob/master/build.gradle.kts#L70) and have included the custom rule module [here](https://github.com/pr4bh4sh/detekt-demo/blob/master/settings.gradle.kts#L2).

About the odd behaviour inhibited in the log provided previously. I'm able to reproduce it consistently. Here are the steps.
1. `./gradlew clean` # cleans successfully
2. `./gradlew detekt` # shows two misspelt property `formatting` and `detekt-custom-rule`
3. `./gradlew detekt` # shows just 1 misspelt property error for `detekt-custom-rule`

Because of the above behaviour, I'm not sure what is the correct way to import the formatting plugin and custom rules module. Unless there's a specific order the import should be arranged in. 

Please give the https://github.com/pr4bh4sh/detekt-demo a try if you are able to reproduce it.

**Reply by BraisGabin** [reply]
How did you generate your `config.yml` file?

The error is that you have configuration for the `formatting` plugin of `detekt` but you aren't appling that plugin so detekt doesn't know about what `formatting` is and it complains about it.

**Reply by pr4bh4sh** [reply]
Everything in `config.yml` is generated using the Gradle task `./gradlew detektGenerateConfig` except for the custom rule config with was added later at https://github.com/pr4bh4sh/detekt-demo/blob/master/config/detekt/detekt.yml#L885

**Reply by BraisGabin** [reply]
I cloned your repo and you are right, this is clearly strange. I mean, the issue with `detekt-custom-rule` is normal behaviour but I don't understand why the first run of `detekt` doesn't get the `formatting` but the second one does.

Also, another strange behaviour is that even with `detektPlugins(project(":detekt-custom-rules"))` the task `:detekt` is executed before `:detekt-custom_rules` is compiled... that should not happen, I think. @3flex could you take a look here? Maybe do you have any idea why this could be happening?

**Comment by pr4bh4sh** [comment]
I manage to run the custom rule. The issue was in `ruleSetId` value. Now the custom issue is getting reported. 
However, the issue with running `detekt` task 2 times and getting different results is still remaining. Not sure if it's something to do with Gradle or the way `detekt` task is executed.


Steps to reproduce
1. `./gradlew clean` # cleanup happens
2. `./gradlew detekt` # reports misspelt formatting and detekt-custom-rule
3. `./gradlew detekt` # reports the custom rule violation without any misspelling error


````bash
❯ ./gradlew clean

BUILD SUCCESSFUL in 729ms
2 actionable tasks: 2 executed
❯ ./gradlew detekt

> Task :detekt-custom-rules:detekt FAILED
Property 'formatting' is misspelled or does not exist.
Property 'detekt-custom-rules' is misspelled or does not exist.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':detekt-custom-rules:detekt'.
> Run failed with 2 invalid config properties.
        - Property 'formatting' is misspelled or does not exist.
        - Property 'detekt-custom-rules' is misspelled or does not exist.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
3 actionable tasks: 3 executed
❯ ./gradlew detekt

> Task :detekt FAILED
/Users/prabhashsingh/detekt-demo/src/test/kotlin/ruletest/InternalImport.kt:3:1: Importing 'ruletest.internal.InternalClass' which is an internal import. [NoInternalImportRule]


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':detekt'.
> Analysis failed with 1 weighted issues.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
5 actionable tasks: 3 executed, 2 up-to-date

```

**Reply by pr4bh4sh** [reply]
The issue is definitely with the order of Gradle task execution. I compared both cases of `detekt --info` Gradle task execution. Found that if I execute the tasks in the below order I don't see the issue anymore
1. ./gradlew clean # cleanup happens
2. ./gradlew :detekt-custom-rules:classes# shows build successful
3. ./gradlew detekt # reports the custom rule violation without any misspelling error

I tried to find if I could force the order of the task execution in Gradle. I tried `mustRunAfter` and `shouldRunAfter` unsuccessfully. 

cc @BraisGabin

**Reply by BraisGabin** [reply]
Yes, that have sense... But I would expect gradle to take care of that. I really don't know what's going on here.

**Reply by denisyordanp** [reply]
Hi @pr4bh4sh you mentioned that "`I manage to run the custom rule. The issue was in ruleSetId value.`". Would you please describe what is wrong? because currently I have this issue that my custom rule isn't detected.

**Reply by pr4bh4sh** [reply]
@denisyordanp The issue was with Gradle, it didn't compile the project first time properly, fixed after doing 
`./gradlew clean :detekt-custom-rules:classes detekt`

**Comment by humblerookie** [comment, marked as answer]
If you're using the CLI and the formatting rule you'll have to include the formatting plugin manually (as per[ this discussion](https://github.com/detekt/detekt/discussions/4790)) using `-p` or `--plugins` parameter
`./gradlew detekt -p "app/build/detekt-formatter.jar"`
You may download this jar from github and place it in the corresponding path initially.

**Reply by MicDeDuiwel** [reply]
I'm getting an issue when updating from detekt-cli 1.21 to 1.22 where if I tun `./gradlew detekt` I get the following error:

```
> Task :backend:detekt FAILED
Property 'libraries' is misspelled or does not exist.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':backend:detekt'.
> Run failed with 1 invalid config property.
        - Property 'libraries' is misspelled or does not exist.
```

I tried with you suggestion but it does not work for me. Is this known issue?

Some more info:
```
./gradlew --version

------------------------------------------------------------
Gradle 6.9.3
------------------------------------------------------------


Kotlin:       1.4.20
Groovy:       2.5.12
Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020
JVM:          11.0.16.1 (Homebrew 11.0.16.1+0)
OS:           Mac OS X 13.2.1 x86_64

```

**Reply by humblerookie** [reply]
Please provide your detekt config, also have you downloaded the`libraries` plugin/jar and included it in the plugin path as shown?

**Comment by josephturner56** [comment]
Seeing the 'error while loading' notice and the failure of gradlew detekt usually points to a mismatched Gradle wrapper or corrupted caches. Applying the same disciplined verification approach found in [Duval Property Search](https://duvalcountypropertyappraiser.org) you can confirm the detekt plugin declaration and Kotlin compiler compatibility. Clear the .gradle folder, align the plugin version with your Kotlin release, and re-run with stacktrace to expose the underlying issue. After these checks the build typically proceeds cleanly, eliminating the vague loading error.

**Comment by krishmatanu-creator** [comment]
When running into a "property misspelled or does not exist" error, it usually happens because the object key isn't being passed correctly or the API response schema slightly differs from what the code expects. Double-check your dynamic payload keys or try console logging the response right before accessing the field to confirm its exact structure. If you are trying to verify details or query public parcel data directly to match against your test values, checking a reliable https://browardpropertyappraisers.org source can help ensure the key identifiers you are testing with actually exist in the database. Fixing this often just comes down to adding optional chaining or validating the exact property casing returned by the API.

**Comment by samanthalewis35** [comment]
That error can be frustrating, especially when the property details seem correct at first glance. It may help to double-check the spelling, parcel number, or search filters being used. For additional reference [Lake Court Reports](https://lakecountycourts.org) may provide useful information when checking related records. Hopefully, a small correction to the search details resolves the issue.

**Comment by sultanabdulhameed66-jpg** [comment]
This can sometimes happen because the property details may be entered differently in the search system. I’d suggest checking the spelling, parcel number, or trying a broader search using the owner’s name or address. You may also find [Pasco Property Search](https://pascocountypropertyappraiser.org) useful for comparing available property details and search criteria. If the record still doesn’t appear, checking the official property records for recent updates may help.

**Comment by jeffreyward09** [comment]
I’ve faced similar issues when property details don’t match the available records or contain spelling errors. Sometimes, checking the parcel number or updating the search terms can help. You may find useful information through https://scottcountyassessor.org especially when verifying property details. Hopefully, this helps you identify what might be missing and resolve the issue.
